frontend/src/scenes/ingestion/panels/BillingPanel.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect } from 'react'
+import { useActions, useValues } from 'kea'
+import { CardContainer } from 'scenes/ingestion/CardContainer'
+import { ingestionLogic } from 'scenes/ingestion/ingestionLogic'
+import { LemonButton } from 'lib/components/LemonButton'
+import './Panels.scss'
+import { eventUsageLogic } from 'lib/utils/eventUsageLogic'
+import BlushingHog from 'public/blushing-hog.png'
+import { BillingEnrollment } from 'scenes/billing/BillingEnrollment'
+import { LemonDivider } from '@posthog/lemon-ui'
+import { IconOpenInNew } from 'lib/components/icons'
+import { billingLogic } from 'scenes/billing/billingLogic'
+import { Plan } from 'scenes/billing/Plan'
+
+export function BillingPanel(): JSX.Element {
+    const { completeOnboarding, setCurrentIndex } = useActions(ingestionLogic)
+    const { reportIngestionContinueWithoutBilling } = useActions(eventUsageLogic)
+    const { billing } = useValues(billingLogic)
+
+    useEffect(() => {
+        if (billing?.plan) {
+            setCurrentIndex(4)
+        }
+    }, [billing])
```

Comment:
This could probably go into ingestionLogic with a [subscription](https://keajs.org/docs/plugins/subscriptions). I find it easier to reason about the flow of things in an app, if all data manipulation logic for one scene is in one file (logic), not split between two (logic + component). 

Here I think the logic could set up a subscription on `billing`, and call the relevant `setCurrentIndex` function there.